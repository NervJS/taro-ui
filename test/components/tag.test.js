import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtTag from '../../.temp/components/tag/index'

describe('AtTag Snap', () => {
  it('render AtNoticebar -- props size', () => {
    const component = renderToString(<AtTag size='small'>标签</AtTag>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props type', () => {
    const component = renderToString(<AtTag type='primary'>标签</AtTag>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props name', () => {
    const component = renderToString(<AtTag name='tag-01'>标签</AtTag>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props circle', () => {
    const component = renderToString(<AtTag circle>标签</AtTag>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props active', () => {
    const component = renderToString(<AtTag active>标签</AtTag>)
    expect(component).toMatchSnapshot()
  })

  it('render AtNoticebar -- props disabled', () => {
    const component = renderToString(<AtTag disabled>标签</AtTag>)
    expect(component).toMatchSnapshot()
  })
})

describe('AtTag Event', () => {
  it('AtTag onClick', () => {
    const onClick = jest.fn()
    const component = renderIntoDocument(<AtTag onClick={onClick}>标签</AtTag>)
    const dom = findDOMNode(component, 'at-tag')
    Simulate.click(dom)
    expect(onClick).toBeCalled()
  })

  it('AtTag onClick not to be called(disabled)', () => {
    const onClick = jest.fn()
    const component = renderIntoDocument(
      <AtTag onClick={onClick} disabled>
        标签
      </AtTag>
    )
    const dom = findDOMNode(component, 'at-tag')
    Simulate.click(dom)
    expect(onClick).not.toBeCalled()
  })

  it('AtTag onClick params {name, active}', () => {
    const onClick = jest.fn()
    const component = renderIntoDocument(
      <AtTag onClick={onClick} name='tag-01'>
        标签
      </AtTag>
    )
    const dom = findDOMNode(component, 'at-tag')
    Simulate.click(dom)
    expect(onClick.mock.calls[0][0].name).toEqual('tag-01')
    expect(onClick.mock.calls[0][0].active).toBeFalsy()
  })
})
