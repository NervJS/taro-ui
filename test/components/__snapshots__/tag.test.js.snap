// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTag Snap render AtNoticebar -- props active 1`] = `"<div class=\\"at-tag at-tag--normal at-tag--active\\">标签</div>"`;

exports[`AtTag Snap render AtNoticebar -- props circle 1`] = `"<div class=\\"at-tag at-tag--normal at-tag--circle\\">标签</div>"`;

exports[`AtTag Snap render AtNoticebar -- props disabled 1`] = `"<div class=\\"at-tag at-tag--normal at-tag--disabled\\">标签</div>"`;

exports[`AtTag Snap render AtNoticebar -- props name 1`] = `"<div class=\\"at-tag at-tag--normal\\">标签</div>"`;

exports[`AtTag Snap render AtNoticebar -- props size 1`] = `"<div class=\\"at-tag at-tag--small\\">标签</div>"`;

exports[`AtTag Snap render AtNoticebar -- props type 1`] = `"<div class=\\"at-tag at-tag--normal at-tag--primary\\">标签</div>"`;
